services/core-api/app/api/mines/reports/models/mine_report.py

```diff
@@ -1,43 +1,97 @@
+    mine_region = association_proxy('mine_table', 'mine_region')
+    major_mine_ind = association_proxy('mine_table', 'major_mine_ind')
+
     permit_id = db.Column(db.Integer, db.ForeignKey('permit.permit_id'))
     permit = db.relationship('Permit', lazy='selectin')
+    permit_guid = association_proxy('permit', 'permit_guid')
+
     received_date = db.Column(db.DateTime)
-    due_date = db.Column(db.DateTime, nullable=False)
+    due_date = db.Column(db.DateTime)
     submission_year = db.Column(db.Integer)
+
     deleted_ind = db.Column(db.Boolean, server_default=FetchedValue(), nullable=False)
 
     mine_report_submissions = db.relationship(
         'MineReportSubmission',
         lazy='joined',
         order_by='asc(MineReportSubmission.mine_report_submission_id)',
-    )
+        uselist=True)
 
     created_by_idir = db.Column(db.String, nullable=False, default=User().get_user_username)
 
-    mine_report_definition_guid = association_proxy('mine_report_definition',
-                                                    'mine_report_definition_guid')
-    report_name = association_proxy('mine_report_definition', 'report_name')
-    report_name = association_proxy('mine_report_definition', 'report_name')
-    permit_guid = association_proxy('permit', 'permit_guid')
+    @hybrid_property
+    def mine_report_submission_status_code(self):
+        if self.mine_report_submissions and len(self.mine_report_submissions) > 0:
+            return self.mine_report_submissions[len(self.mine_report_submissions) -
+                                                1].mine_report_submission_status_code
+        else:
+            return None
+
+    @mine_report_submission_status_code.expression
+    def mine_report_submission_status_code(cls):
+        return select([
+            MineReportSubmission.mine_report_submission_status_code
+        ]).where(MineReportSubmission.mine_report_id == cls.mine_report_id).order_by(
+            desc(MineReportSubmission.mine_report_submission_id)).limit(1).as_scalar()
+
+    @hybrid_property
+    def mine_report_submission_status_code_description(self):
+        if self.mine_report_submissions and len(self.mine_report_submissions) > 0:
+            return MineReportSubmissionStatusCode.find_by_mine_report_submission_status_code(
+                self.mine_report_submission_status_code).description
+        else:
+            return None
+
+    @mine_report_submission_status_code_description.expression
+    def mine_report_submission_status_code_description(cls):
+        return select([MineReportSubmissionStatusCode.description]).where(
+            and_(
+                MineReportSubmission.mine_report_id == cls.mine_report_id,
+                MineReportSubmissionStatusCode.mine_report_submission_status_code ==
+                MineReportSubmission.mine_report_submission_status_code)).order_by(
+                    desc(MineReportSubmission.mine_report_submission_id)).limit(1).as_scalar()
```

Comment:
same naming concern. Please look into defining a relationship for the most recent mine_report_submission, that might be better performance and easier to follow than this set of functions.

Reply:
I tried to do that and I couldn't find a way to create a property that just gets the last element, so I opted to create a hybrid property instead.

Reply:
Resolved in person.

Reply:
Maybe a comment on why these exists, After explanation i understand. We a sqlalchemy to sort/filter on these members, but they don't exist in the db, so we have these expressions to tell the ORM how to execute operations on these members.

I have no idea around the performance implications.